packages/jest-config/src/normalize.js

```diff
@@ -113,24 +113,20 @@ const setupBabelJest = (options: InitialOptions) => {
     });
 
     if (customJSPattern) {
-      const jsTransformer = Resolver.findNodeModule(
```

Comment:
You are removing the resolution for packages that are not `babel-jest`, like `ts-jest` or `./my-custom-file` or `my-custom-transformer`. You still need to support those as well.

Reply:
Hm, I see what you mean. Do you think it would be reasonable to have the same `findNodeModule` call as a fallback, but encourage people to use `require.resolve` in their configuration? ie

```js
module.exports = {
  transform: {
    "*.ts": require.resolve('ts-jest')
  }
};
```

An issue is that it wouldn't work as-is in the package.json configuration, since functions can't be called here :/

Reply:
An alternative would be to defer resolving those paths until after the configuration has been completed - but the same issue will occur for the `resolver` option, which ironically will need to be manually resolved through `require.resolve`.

Reply:
What if in Jest we call `Resolver.findNodeModule(value)`. If it returns null, we call `Resolver.findNodeModule(require.resolve(value))` (or just `require.resolve(value)` itself if possible). That way it doesn't affect users.

Reply:
But then it will resolve them relative to jest-config, not the user project. So `ts-jest` for example could fail because it would not be listed as a dependency of `jest-config` (whereas a `require.resolve` from the configuration file would always work, because the configuration file would be in the toplevel project).

Reply:
Actually I just read the code again and I don't think this PR prevents external transformers from working - `setupBabelJest` is only used to detect babel jest and, if it can be found, automatically inject `regenerator`. Any other loader will be resolved and required later, through the regular resolver (cf [`script_transformer.js`](https://github.com/facebook/jest/blob/master/packages/jest-runtime/src/script_transformer.js)).